concent_api/core/validation.py

```diff
@@ -208,59 +182,39 @@ def is_golem_message_signed_with_key(
 
 def validate_golem_message_subtask_results_rejected(subtask_results_rejected: message.tasks.SubtaskResultsRejected):
     if not isinstance(subtask_results_rejected,  message.tasks.SubtaskResultsRejected):
-        raise Http400(
+        raise ConcentValidationError(
             "subtask_results_rejected should be of type:  SubtaskResultsRejected",
             error_code=ErrorCode.MESSAGE_INVALID,
         )
     validate_task_to_compute(subtask_results_rejected.report_computed_task.task_to_compute)
 
 
-def validate_subtask_price_task_to_compute(task_to_compute: message.tasks.TaskToCompute):
-    if not isinstance(task_to_compute.price, int):
-        raise Http400(
-            "Price must be a integer",
-            error_code=ErrorCode.MESSAGE_VALUE_NOT_INTEGER,
-        )
-    if task_to_compute.price < 0:
-        raise Http400(
-            "Price cannot be a negative value",
-            error_code=ErrorCode.MESSAGE_VALUE_NEGATIVE,
-        )
-
-
 def validate_ethereum_addresses(requestor_ethereum_address, provider_ethereum_address):
     if not isinstance(requestor_ethereum_address, str):
-        raise Http400(
+        raise ConcentValidationError(
```

Comment:
Those 4 checks could be paired and checked in `validate_key_with_desired_parameters`.